query-algebrizer/src/cc.rs

```diff
@@ -534,8 +531,11 @@ impl ConjoiningClauses {
     /// `is_known_empty`.
     fn alias_table<'s, 'a>(&mut self, schema: &'s Schema, pattern: &'a Pattern) -> Option<SourceAlias> {
```

Comment:
I only see one consumer of `alias_table`, and it immediately checks for `None` and marks the CC as known empty (potentially for a different reason).  So: should this just be `Result` the whole way through, rather than introducing `Option` and then reverting to `Result`?  And do you expect to want to _chain_ `EmptyBecause` reasons to provide more feedback?